internal/vm/setup.go

```diff
@@ -74,6 +73,114 @@ _G["load_plugin_schema"] = function(plugin_schema_string)
     return nil, "error initializing schema for plugin: " .. err
   end
 end
+
+-- Remove functions from a schema definition so that
+-- cjson can encode the schema.
+-- Taken from kong/api/api_helpers.lua
+local schema_to_jsonable
+do
+  local insert = table.insert
+  local ipairs = ipairs
+  local next = next
+  local cjson = { array_mt = {} } --- TODO(hbagdi) XXX analyze the impact
```

Comment:
Copied this from the patch and kept the `TODO` label for you @hbagdi; I can remove this if you find this unnecessary.

Reply:
This TODO is fine.
Does this actually do the trick we need it to do? Mainly, that the arrays are rendered as they should be.
Can we please add tests to ensure that?

Reply:
Yep, this was on my list of things to add; a full test which ensures fields arrays and runs the gamut.